var/spack/repos/builtin/packages/qscintilla/package.py

```diff
@@ -98,57 +123,90 @@ def make_qsci(self):
-                    "--qsci-sipdir=" + os.path.join(self.prefix.share.sip, pyqtx),
-                    "--apidir=" + self.prefix.share.qsci,
-                    "--destdir=" + pydir,
-                    "--pyqt-sipdir=" + os.path.join(self.spec[py_pyqtx].prefix.share.sip, pyqtx),
-                    "--sip-incdir="
-                    + join_path(
-                        self.spec["py-sip"].prefix.include,
-                        "python" + str(self.spec["python"].version.up_to(2)),
-                    ),
-                    "--stubsdir=" + pydir,
-                )
-
+            with working_dir(join_path(self.stage.source_path, "src")):
                 # Fix build errors
                 # "QAbstractScrollArea: No such file or directory"
                 # "qprinter.h: No such file or directory"
                 # ".../Qsci.so: undefined symbol: _ZTI10Qsci...."
-                qscipro = FileFilter("Qsci/Qsci.pro")
+                qscipro = FileFilter("qscintilla.pro")
                 if "^qt@4" in self.spec:
                     qtx = "qt4"
                 elif "^qt@5" in self.spec:
                     qtx = "qt5"
+                elif "^qt-base@6" in self.spec:
+                    qtx = "qt6"
+
 
                 link_qscilibs = "LIBS += -L" + self.prefix.lib + " -lqscintilla2_" + qtx
                 qscipro.filter(
                     "TEMPLATE = lib",
-                    "TEMPLATE = lib\nQT += widgets" + "\nQT += printsupport\n" + link_qscilibs,
+                    "TEMPLATE = lib\nINCLUDEPATH += {includepath}\nQT += widgets" + "\nQT += printsupport\n" + link_qscilibs,
                 )
 
                 make()
 
                 # Fix installation prefixes
                 makefile = FileFilter("Makefile")
-                makefile.filter(r"\$\(INSTALL_ROOT\)", "")
-                makefile = FileFilter("Qsci/Makefile")
-                makefile.filter(r"\$\(INSTALL_ROOT\)", "")
+                makefile.filter("$(INSTALL_ROOT)", "", string=True)
+                #makefile = FileFilter("Qsci/Makefile")
+                #makefile.filter("$(INSTALL_ROOT)", "", string=True)
 
                 if "@2.11:" in self.spec:
                     make("install", parallel=False)
                 else:
-                    make("install")
+                    make("install", parallel=False)
+
+            if 'py-pyqt6' in self.spec:
+                with working_dir(join_path(self.stage.source_path, "Python")):
+                    cp = which('cp')
+                    cp('pyproject-qt6.toml', 'pyproject.toml')
+                    # TODO below sip_inc_dir is incorrect:
+                    # its prefix of qscintilla itself as opposed to prefix for py-pyqt6
+                    # qscintilla+python builds fine when sip_inc_dir is hardcoded!
+                    str(self.spec)
+                    sip_inc_dir = join_path(self.spec['py-pyqt6'].prefix, python_platlib, 'PyQt6', 'bindings' )
```

Comment:
@adamjstewart This is driving me crazy. `self.spec['py-pyqt6'].prefix` returns qscintilla's prefix instead.

Reply:
That doesn't make any sense. I'll investigate when I get a chance.

Reply:
I've solved about half of the mystery. I added the following print statements:
```python
print(self.spec['py-pyqt6'].prefix)
print(python_platlib)
print(sip_inc_dir)
```
The output is:
```
/Users/Adam/spack/opt/spack/darwin-ventura-m2/apple-clang-14.0.3/py-pyqt6-6.5.1-iylw7hu6frfmltzjvyfrcorwtkql6s5t
/Users/Adam/spack/opt/spack/darwin-ventura-m2/apple-clang-14.0.3/qscintilla-2.13.3-qlvmxmx7sa447biexlvd26dlxjhp5tih/lib/python3.10/site-packages
/Users/Adam/spack/opt/spack/darwin-ventura-m2/apple-clang-14.0.3/qscintilla-2.13.3-qlvmxmx7sa447biexlvd26dlxjhp5tih/lib/python3.10/site-packages/PyQt6/bindings
```
The first two are expected. I have no idea how `os.path.join` combined 1 and 2 to get 3, but I think what you want is:
```suggestion
                    sip_inc_dir = join_path(self.spec['py-pyqt6'].prefix, self.spec['python'].package.platlib, 'PyQt6', 'bindings' )
```

Reply:
it works! Thanks